### ISSUE TITLE
Compression messing up Server-Sent Events

### ISSUE THREAD

**Issue by Timtam** [Ameobea/rocket_async_compression#12]
Hello,

I was trying to use rocket_async_compression in my project and it worked fine, at least until I discovered that it seems to break my SSEs in some way. I don't know what exactly happened, but I implemented it as described within the official rocket 0.5 chat example and it works just fine without Compression::fairing() attached, and stops working as soon as I attach it. I'd expect this fairing to not interfere with SSEs as they don't support compression (although I'm not sure about that), but it seems to cause some trouble here. Would you be able to check that? Thanks!

**Comment by Ameobea** [comment]
Hi,

I was indeed able to replicate this issue.  I did some looking online, and it seems that compression for SSE event streams is a bit of a tricky case with several webservers and other software having reported issues with compression on them.

To be 100% honest, I don't know what the best way to handle compression for them is.  To deal with the issue you were running into, I decided to disable compression on the `text/event-stream` MIME type which is used by server-sent events.  I've pushed out a new version of `rocket_async_compression` (version 0.6.0) and tested that the official Rocket SSE chat example now works when using the new version.

This will leave the SSE responses uncompressed, but allow compression to continue working on other responses like static HTML/JS, JSON, etc.

In addition to disabling compression for `text/event-stream` responses by default, I also added the ability to manually configure the list of excluded MIME types: https://docs.rs/rocket_async_compression/latest/rocket_async_compression/struct.Compression.html#method.exlude_content_types

I hope this will help with your use case and hopefully prevent others from running into similar issues in the future as well.
